libraries/azure_ad_users.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'azurerm_resource'
+
+class AzureAdUsers < AzurermResource
+  name 'azure_ad_users'
+  desc 'Verifies settings for a collection of Azure Active Directory Users'
+  example "
+    describe azure_ad_users do
+        it                  { should exist }
+    end
+  "
+
+  FilterTable.create
+      .add_accessor(:entries)
+      .add_accessor(:where)
+      .add(:exists?)        { |obj| !obj.entries.empty? }
+      .add(:object_ids,     field: 'objectId')
+      .add(:display_names,  field: 'displayName')
+      .add(:mails,          field: 'mail')
+      .add(:user_types,     field: 'userType')
+      .connect(self, :table)
+
+  attr_reader(:table)
+
+  def initialize
+
+    user_rows = []
+    next_page = nil
+
+    @users = graph_client.get_users
+    return if @users.nil? || @users.empty?
+
+    loop do # Users may be paginated
+      if next_page != nil # Skip in first iteration
+        @users = graph_client.get_users_next(next_page)
+      end
+
+      @users["values"].map do |user|
+        user_rows += [{
+                          objectId:     user["objectId"],
+                          displayName:  user["displayName"],
+                          mail:         user["mail"],
+                          userType:     user["userType"]
+                      }]
+      end
+      next_page = @users["nextLink"]
+      break unless next_page
+    end
+    @table = user_rows
+  end
```

Comment:
Let's revisit this code after the pagination refactoring discussed elsewhere. Long & short of it is that the above should be something more like:
```ruby
def initialize
  @table = graph_client.users
end
```
_Maybe_
```ruby
USER_KEYS = %i(objectId displayName mail userType).freeze

# ... FilterTable and whatnot

def initialize
  @table = graph_client.users.map(&user_to_hash)
end

def user_to_hash
  ->(u) { USER_KEYS.zip(u.values_at(USER_KEYS.map(&:to_s)).to_hash) }
end
```
...but I believe this is actually contrary to the established pattern, where we've kept parity with the AWS implementation and collections of resources are simply lists of identifiers.

Reply:
Sure - I totally agree the pagination should be dealt with at a level below this, I just wasn't sure where best suited. Can you link me to the previous discussions mentioned so I can get up to speed? Will leave how it is until we have chatted and I have a clearer picture of where the logic belongs.